src/averageAges.js

```diff
@@ -36,8 +45,18 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(({ sex }) => sex === 'f');
+
+  const ecssentialWomenAges = !withChildren
+    ? women.map(({ born, died }) => died - born)
+    : women.filter(({ name }) =>
+      people.find(({ mother }) => mother === name))
+      .map(({ born, died }) => died - born);
```

Comment:
```suggestion
    : women.filter(({ name }) =>
      people.find(({ mother }) => mother === name))
      .map(({ born, died }) => died - born);
```

I get lost in this code and I can't tell where the filter ends and which array you use map for. It needs to be formatted